MASH-FRET/source/mod-trace-processing/management/traceManager.m

```diff
@@ -1181,11 +1197,214 @@ function openMngrTool(h_fig)
+    'fontweight','bold');
+
+yNext = yNext - mg - h_txt - h_edit;
+
+h.tm.edit_xrangeLow = uicontrol('style','edit','parent',...
+    h.tm.uipanel_range,'string','0','position',...
+    [xNext,yNext,w_edit,h_edit],'fontunits','pixels','fontsize',fntS,...
+    'callback',{@edit_xrangeLow_Callback,h_fig});
+
+yNext = yNext + h_edit;
+
+h.tm.text_xrangeLow = uicontrol('style','text','parent',...
+    h.tm.uipanel_range,'string','xlow','position',...
+    [xNext,yNext,w_edit,h_txt],'fontunits','pixels','fontsize',fntS,...
+    'horizontalalignment','center');
+
+yNext = yNext - h_edit;
+xNext = xNext + w_edit + mg/2;
+
+h.tm.edit_xrangeUp = uicontrol('style','edit','parent',...
+    h.tm.uipanel_range,'string','1','position',...
+    [xNext,yNext,w_edit,h_edit],'fontunits','pixels','fontsize',fntS,...
+    'callback',{@edit_xrangeUp_Callback,h_fig});
+
+yNext = yNext + h_edit;
+
+h.tm.text_xrangeUp = uicontrol('style','text','parent',...
+    h.tm.uipanel_range,'string','xup','position',...
+    [xNext,yNext,w_edit,h_txt],'fontunits','pixels','fontsize',fntS,...
+    'horizontalalignment','center');
+
+yNext = yNext - h_edit;
+xNext = xNext + w_edit + mg;
+
+h.tm.edit_yrangeLow = uicontrol('style','edit','parent',...
+    h.tm.uipanel_range,'string','0','position',...
+    [xNext,yNext,w_edit,h_edit],'fontunits','pixels','fontsize',fntS,...
+    'callback',{@edit_yrangeLow_Callback,h_fig},'enable','off');
+
+yNext = yNext + h_edit;
+
+h.tm.text_yrangeLow = uicontrol('style','text','parent',...
+    h.tm.uipanel_range,'string','ylow','position',...
+    [xNext,yNext,w_edit,h_txt],'fontunits','pixels','fontsize',fntS,...
+    'horizontalalignment','center','enable','off');
+
+yNext = yNext - h_edit;
+xNext = xNext + w_edit + mg/2;
+
+h.tm.edit_yrangeUp = uicontrol('style','edit','parent',...
+    h.tm.uipanel_range,'string','1','position',...
+    [xNext,yNext,w_edit,h_edit],'fontunits','pixels','fontsize',fntS,...
+    'callback',{@edit_yrangeUp_Callback,h_fig},'enable','off');
+
+yNext = yNext + h_edit;
+
+h.tm.text_yrangeUp = uicontrol('style','text','parent',...
+    h.tm.uipanel_range,'string','yup','position',...
+    [xNext,yNext,w_edit,h_txt],'fontunits','pixels','fontsize',fntS,...
+    'horizontalalignment','center','enable','off');
+
+xNext = mg;
+yNext = yNext - h_edit - mg - h_txt;
+
+h.tm.text_conf1 = uicontrol('style','text','parent',h.tm.uipanel_range,...
+    'string','Confidence:','position',...
+    [xNext,yNext,w_pan_slct-2*mg,h_txt],'fontunits','pixels','fontsize',...
+    fntS,'horizontalalignment','left','fontweight','bold');
+
+yNext = yNext - h_txt;
+
+h.tm.text_conf2 = uicontrol('style','text','parent',h.tm.uipanel_range,...
+    'string','(for trajectories only)','position',...
+    [xNext,yNext,w_pan_slct-2*mg,h_txt],'fontunits','pixels','fontsize',...
+    fntS,'horizontalalignment','left','fontangle','italic');
+
+yNext = yNext - mg - h_pop;
+
+str_pop = {'confidence in percent','confidence in data points'};
```

Comment:
The range bounds combined with the confidence settings is a really nice feature to speed up the molecule selection! Since the confidence setting is a bit harder to grasp I suggest the following small rearrangement:
Put the drop down menu that now reads `confidence in percent` / `confidence in data points` beneath the `at least` / `at most` / `between` drop down menu, such that it reads as follows: "**select trace if** `between` `50` **and** `100` `percent of the trace` / `data points` **are within xlow and xup**".

Reply:
That's a good idea, thank you, it is implemented in MASH-code 6

Reply:
On top of the rearrangment I would explicitely put the sentence "**select trace if** `between` `50` **and** `100` `percent of the trace` / `data points` **are within xlow and xup**".
**This** is static text and `this` are the preexisting edit fields or drop down menus (i.e. adjust the string in the drop down from "confidence in percent" to [...] "percent of the trace" and "confidence in data points" to [...] "data points" and then add another static text: [...] **are within xlow and xup**. But I leave it up to you.

Reply:
Ok I get it now! I've added **Select trace if** [..] **are included in the range**

Reply:
👍 Cool, thanks!